Meta/Lagom/Tools/ELFBaker/main.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Error.h>
+#include <AK/String.h>
+#include <AK/StringView.h>
+#include <AK/Vector.h>
+#include <LibC/elf.h>
+#include <LibCore/File.h>
+#include <LibMain/Main.h>
+
+template<typename ElfHeader, typename ElfPhdr, typename ElfShdr>
+class ElfImage : public ByteBuffer {
```

Comment:
This inheritance feels strange, can we decouple these concepts?
Also `ElfImage` kinda clashes with `ELF::Image` which is something in LibELF.